### ISSUE TITLE
The query got a bad result in V2.18.1

### ISSUE THREAD

**Issue by lixy529** [prometheus/prometheus#7349]
```
remote_write:
  - url: "http://127.0.0.1:9201/write"
remote_read:
  - url: "http://127.0.0.1:9201/read"
  read_recent: true
```

In v2.18.1, remote_storage is used to read data. When merging data, value is always the last value.
for example:
input:
badtestprom{cluster="h01"} 3
badtestprom{cluster="h02"} 4
got:
badtestprom{cluster="h01",job="lixytest"} 4
badtestprom{cluster="h02",job="lixytest"} 4
expect:
badtestprom{cluster="h01",job="lixytest"} 3
badtestprom{cluster="h02",job="lixytest"} 4

**Comment by roidelapluie** [comment]
It is difficult to know if it is a bug with prometheus or the remote read endpoint you are using. You could run the query: badtestprom[5m] to find out. Also, you could remove read_recent: true.

**Comment by lixy529** [comment]
If I query through badtestprom{cluster="h01"}, the result is correct.

**Comment by lixy529** [comment]
It should be that when merging the results, the last value will overwrite the previous value.

**Comment by roidelapluie** [comment]
So this is 2.18.1 not 2.8.1?

see https://github.com/prometheus/prometheus/issues/7359

**Comment by csmarchbanks** [comment]
This was fixed in #7361, and deployed to both v2.18.2 and v2.19.0. If you are still seeing the incorrect behavior in those versions please let us know!
